Release notes for DTM 10.17 version (#145)
SoftwareAG/c8y-releasenotes: content/release-10-17-0/digital-twin-manager-10-17-0-bundle/10-17-0-0.md

---
weight: 40
title: Release 10.17.0.0
layout: bundle
---

### Improvements

<div><table ><colgroup>
<col style="width: 70%;"><col style="width: 15%;"></colgroup>
<thead><tr>
<th>
Description</th>
<th>
Issue</th>
</tr>
</thead><tbody>

<tr>
<td>DTM application is now added with the <b>Home</b> page which mainly provides brief information about the application along with navigation to User guide. User can even use this page to navigate to <b>Asset properties, Asset models</b> and <b>Assets</b> creation page.</td>
<td>CTM-311</td>
</tr>

<tr>
<td><b>Assets</b> page now displays tree view representation of the assets for better visualize large hierarchies.</td>
<td>CTM-128</td>
</tr>

<tr>
<td>On click of an asset now user is provisioned to view the hierarchical/tree view representation of the asset under <b>Asset tree</b> tab. User can even expand and collapse the view using <b>Expand all</b> and <b>Collapse all</b> button.</td>
<td>CTM-129</td>
</tr>

<tr>
<td>Search feature is now added to the <b>Assets</b> page.</td>
<td>CTM-227</td>
</tr>

<tr>
<td>Default <b>Group</b> creation in <b>Asset models</b> grid is removed.</td>
<td>CTM-388</td>
</tr

<tr>

> <td>Assets navigation from left navigation menu is no more supported.</td>

Reviewer: ```suggestion
<td>Navigating to view details of an asset using left navigation is removed.</td>
```

<td>CTM-389</td>
</tr

<tr>

> <td>Assets can be added with child devices now.</td>

Reviewer: ```suggestion
<td>Child devices of a device can also be assigned to an asset.</td>
```

<td>CTM-316</td>
</tr>

<tr>

> <td>Display of devices in the <b>Assets</b> and <b> Asset tree</b> page has been restricted.</td>

Reviewer: ```suggestion
<td>Displaying devices assigned to an asset, is restricted in hierarchical/tree-view representation in the <b>Assets</b> and <b> Asset tree</b> pages.</td>
```

<td>CTM-422</td>
</tr>

<tr>

> <td>Added a confirmation popup which states that the changes made to the <b>Asset hierarchy</b> page will be lost while leaving the page abruptly.</td>

Reviewer: ```suggestion
<td>A confirmation popup which states that the changes made to the <b>Asset hierarchy</b> will be lost, is shown when a user leaves the page abruptly.</td>
```

<td>CTM-249</td>
</tr>

<tr>

> <td>Added a hint which indicates the field is required while creating the asset.</td>

Reviewer: Does the hint have text to indicate the field is required?

Author: Yes.. Example: <<AssetProperyName>>_(required)_

<td>CTM-248</td>
</tr>

<tr>

> <td>Creation of assets with deep hierarchy will display parent hierarchical details in user convenient way.</td>

Reviewer: Expand on what is the convenient way to the user as this sounds vague.

<td>CTM-250</td>
</tr>

<tr>

> <td><b>Add child asset model</b> drop down now displays description along with asset model name.</td>

Reviewer: ```suggestion
<td>In the <b>New asset model</b> page, <b>Add child asset model</b> drop down displays description along with asset model name.</td>
```

<td>CTM-236</td>
</tr>

<tr>

> <td><b>Add asset property</b> drop down now displays description along with asset property name.</td>

Reviewer: ```suggestion
<td>In the <b>New asset model</b> page, <b>Add asset property</b> drop down displays description along with asset property name.</td>
```

<td>CTM-237</td>
</tr>

<tr>
<td>Delight 2.0 branding changes has been incorporated to DTM.</td>
<td>CTM-162</td>
</tr>

<tr>
<td>Sorting and filtering is now available for <b>Model name</b> column in <b>Asset models</b> page.</td>
<td>CTM-252</td>
</tr>

<tr>
<td>Nomenclature for <b>Asset types</b> and <b>Property library</b> is now changed to <b>Asset Models</b> and <b>Asset properties</b> respectively. Left navigation menu items are updated with respect to their positions along with the modified nomenclature.</td>
<td>CTM-253</td>
</tr>

<tr>

> <td><b>Asset models</b> and <b>Asset properties</b> are now implemented with auto key generation during their creation.</td>

Reviewer: ```suggestion
<td>During the creation of <b>Asset models</b> and <b>Asset properties</b>, the Key field is automatically populated with the value based on the label.</td>
```

<td>CTM-235</td>
</tr>

<tr>
<td>DTM microservice now uses API version 2.</td>
<td>CTM-169</td>
</tr>

<tr>
<td>Focus is now set on the <b>Name</b> field while creating the asset.</td>
<td>CTM-238</td>
</tr>

<tr>

> <td>Nesting of Asset models is now handled by blocking during Asset model creation.</td>

Reviewer: ```suggestion
<td>All the parent asset models in the hierarchy of the current asset model are not listed in the <b>Add child asset model</b> selection to restrict cyclical nesting of asset models.</td>
```

<td>CTM-239</td>
</tr>

</tbody></table></div>

### Fixes

<div><table ><colgroup>
<col style="width: 70%;"><col style="width: 15%;"></colgroup>
<thead><tr>
<th>
Description</th>
<th>
Issue</th>
</tr>
</thead><tbody>

<tr>

> <td>Issues related to the <b>Assets</b> page has been addressed.</td>

Reviewer: Elaborate in few words what the issues are.

<td>CTM-156</td>
</tr>

<tr>

> <td>Asset properties will display after creation of an asset.</td>

Reviewer: ```suggestion
<td>Asset properties are displayed in the asset details after creation of an asset.</td>
```

<td>CTM-322</td>
</tr>

<tr>

> <td>Issues observed in <b>Assets</b> and <b>Asset tree</b> page are fixed.</td>

Reviewer: Elaborate in few words what the issues are.

<td>CTM-382</td>
</tr>

<tr>

> <td>Issues observed in <b>Home</b> page are fixed.</td>

Reviewer: Elaborate in few words what the issues are.

<td>CTM-403</td>
</tr>

<tr>

> <td>Filter applied on <b>Asset models</b> grid will now get saved after changing tab.</td>

Reviewer: ```suggestion
<td>Filter applied on <b>Asset models</b> grid is saved even after changing tabs.</td>
```

<td>CTM-324</td>
</tr>

<tr>

> <td>Fixed the navigation issue related to <b>Asset tree</b> page.</td>

Reviewer: Describe the issue in few words

<td>CTM-406</td>
</tr>

<tr>
<td>Issue related to child hierarchy creation in <b>Asset tree</b> page is fixed.</td>
<td>CTM-405</td>
</tr>

<tr>

> <td>Button name in <b>Assign devices</b> window is changed to <b>Enable child devices selection</b>.</td>

Reviewer: Mention what the previous name was.
```suggestion
<td>Button name in <b>Assign devices</b> dialog is changed to <b>Enable child devices selection</b>.</td>
```

<td>CTM-409</td>
</tr>

<tr>

> <td>Fixed an issue where in Asset search failed if there are multiple instances of an asset.</td>

Reviewer: ```suggestion
<td>Fixed failing Asset search if there are multiple instances of an asset.</td>
```

> <td>Filter applied on <b>Asset models</b> grid is saved even after changing tabs.</td>

Author: Changed grid to columns

<td>CTM-331</td>
</tr>

<tr>

> <td>Fixed the abrupt closure of <b>Assign devices</b> window after child devices is assigned.</td>

Reviewer: ```suggestion
<td>Fixed abrupt closure of <b>Assign devices</b> dialog after child devices are assigned.</td>
```

<td>CTM-332</td>
</tr>

<tr>

> <td><b>Import assets</b> button will not get displayed if micro service is down.</td>

Reviewer: ```suggestion
<td><b>Import assets</b> button will not be displayed if micro service is not available.</td>
```

<td>CTM-419</td>
</tr>

<tr>
<td>Delay in the display of message in <b>Localization</b> page is addressed.</td>
<td>CTM-425</td>
</tr>

<tr>

> <td>Fixed the issue related to the display of message in <b>Assets model</b> page.</td>

Reviewer: Elaborate what the issue is in few words

<td>CTM-426</td>
</tr>

<tr>

> <td>Translation is now handled in the popup message on deletion of asset property from the backend.</td>

Reviewer: ```suggestion
<td>Translation of the message shown in the confirmation popup on deletion of asset property using Inventory API is added.</td>
```

<td>CTM-421</td>
</tr>

<tr>

> <td>Microservice start up issue is now fixed.</td>

Reviewer: ```suggestion
<td>Issue of microservice logging errors on startup and having the status as 'Unhealthy' is resolved.</td>
```

> <td>Fixed abrupt closure of <b>Assign devices</b> dialog after child devices are assigned.</td>

Author: Committed the suggestion

<td>CTM-432</td>
</tr>

<tr>

> <td>Issues observed in <b>Asset properties</b> page is fixed.</td>

Reviewer: Elaborate in few words what the issues are.

> <td><b>Import assets</b> button will not be displayed if micro service is not available.</td>

Author: Committed the suggestion

<td>CTM-424</td>
</tr>

<tr>

> <td>Fixed an issue where in bulk import failed for file property.</td>

Reviewer: ```suggestion
<td>Fixed the issue of failing bulk import when an asset property of type 'File upload' was present.</td>
```

> <td>Delay in the display of message in <b>Localization</b> page is addressed.</td>

Author: Committed the suggestion

<td>CTM-418</td>
</tr>

<tr>

> <td><b>Assets</b> page is now shown with a message if there are no asset exists.</td>

Reviewer: ```suggestion
<td><b>Assets</b> page shows a message if there are no assets created.</td>
```

> <td>Message is displayed in the <b>Assets model</b> page if there are no Asset models created.</td>

Author: Committed the suggestion

> <td>Translation of the message shown in the confirmation popup on deletion of asset property using Inventory API is added.</td>

Author: Committed the suggestion

> <td>Issue of microservice logging errors on startup and having the status as 'Unhealthy' is resolved.</td>

Reviewer 2: ```suggestion
<td>Microservice logging errors on startup are now resolved and the status is no longer shown as "Unhealthy".</td>
```

Author: Committed the suggestion

> <td>Message is added in <b>Asset properties</b> page if there are no asset properties created.</td>

Reviewer 2: ```suggestion
<td>A message is now shown in the <b>Asset properties</b> page if there are no asset properties available.</td>
```

Author: Committed the suggestion

> <td>Fixed the issue of failing bulk import when an asset property of type 'File upload' was present.</td>

Reviewer 2: ```suggestion
<td>The bulk import no longer fails in case of an asset property of type "File upload".</td>
```

Author: Committed the suggestion

> <td><b>Assets</b> page shows a message if there are no assets created.</td>

Reviewer 2: ```suggestion
<td>A message is now shown in the <b>Assets</b> page if there are no assets available.</td>
```

Author: Committed the suggestion
